temporal-object/src/main/java/com/iluwatar/temporalobject/App.java

```diff
@@ -0,0 +1,76 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.iluwatar.temporalobject;
+
+/**
+ * The Temporal Object pattern is an object that changes over time.
+ *
+ * <p>There are times where it is useful to consider that an object that has
+ * properties that change over time, and others where the object is temporal. </p>
+ *
+ * <p>In this particular example, each {@link CreditCard} has a {@link CreditContractVersionHistory},
+ * which stores the prior and current contract terms as {@link CreditContractVersion}s. In this case,
+ * most details about the credit card are part of the contract, and so, the {@link CreditCard} is
+ * considered a Temporal Object.</p>
+ */
+public class App {
+  /**
+   * Main function.
+   *
+   * @param args Unused arguments.
+   */
+  public static void main(String[] args) {
+    SimpleDate.setToday(new SimpleDate(2000, 3, 1));
+
+    // start a credit card contract from today until 3rd of September 2002
+    CreditContractVersion firstContract = new CreditContractVersion("Original Contract",
+            "Bank of XYZ", "Version 1", 100, 12345, 1234,
+            new SimpleDate(2002, 9, 3));
+    CreditCard card = new CreditCard(firstContract, SimpleDate.getToday());
+    System.out.println("New card with limit of " + card.getCreditLimit()
+            + ", card number " + card.getNumber() + ", and cvc code of " + card.getCvc()
+            + " which expires on " + card.getExpiration());
+
+    // The card expires
+    SimpleDate.setToday(new SimpleDate(2002, 9, 18));
+    System.out.println(SimpleDate.getToday().toString() + " is expired: " + card.isExpired());
+
+    // And is then re-negotiated, with the new contract coming into effect a few days later
+
+    CreditContractVersion secondContract = new CreditContractVersion("new contract", "Bank of XYZ",
+            "Version 2", 120, 12345, 1234, new SimpleDate(2004, 3, 4));
+    card.addContract(secondContract, new SimpleDate(2002, 9, 21));
+
+    // A few days later when the new contract is effective
+    SimpleDate.setToday(new SimpleDate(2002, 9, 21));
+    System.out.println("Re-negotiated card with limit of " + card.getCreditLimit()
+            + ", card number " + card.getNumber() + ", and cvc code of " + card.getCvc()
+            + " which expires on " + card.getExpiration());
+    System.out.println(SimpleDate.getToday().toString() + " is expired: " + card.isExpired());
```

Comment:
Other examples in the repository tend to use `LOGGER.info` instead of `sout`. See: https://github.com/iluwatar/java-design-patterns/tree/master/factory